.github/workflows/test.yml

```diff
@@ -8,11 +8,20 @@ on:
 
 # Current configuration:
 # - django 3.2, python 3.8, wagtail 4.1, postgres
-# - django 4.0, python 3.9, wagtail 4.2, sqlite
-# - django 4.1, python 3.10, wagtail 5.0, postgres
-# - django 4.2, python 3.11, wagtail 5.0, sqlite
-# - django 4.1, python 3.10, wagtail 5.1, postgres
-# - django 4.2, python 3.11, wagtail 5.1, sqlite
+# - django 4.1, python 3.8, wagtail 4.1, sqlite
+# - django 3.2, python 3.8, wagtail 5.2, postgres
+# - django 4.1, python 3.8, wagtail 5.2, sqlite
+# - django 3.2, python 3.9, wagtail 4.1, postgres
+# - django 4.1, python 3.9, wagtail 4.1, sqlite
+# - django 3.2, python 3.9, wagtail 5.2, postgres
+# - django 4.1, python 3.9, wagtail 5.2, sqlite
+# - django 3.2, python 3.10, wagtail 4.1, postgres
+# - django 4.1, python 3.10, wagtail 4.1, sqlite
+# - django 3.2, python 3.10, wagtail 5.2, postgres
+# - django 4.1, python 3.10, wagtail 5.2, sqlite
+# - django 4.1, python 3.11, wagtail 4.1, sqlite
+# - django 4.1, python 3.11, wagtail 5.2, sqlite
+# - django 4.2, python 3.11, wagtail 5.2, sqlite
 # - django 4.2, python 3.11, wagtail main, sqlite (allow failures)
```

Comment:
I think we can safely trim this down a lot further (and add Python 3.12 while we're at it). I'd suggest:

```
# - django 3.2, python 3.8, wagtail 4.1, postgres
# - django 3.2, python 3.9, wagtail 5.1, sqlite
# - django 4.1, python 3.10, wagtail 5.1, postgres
# - django 4.2, python 3.11, wagtail 5.2, sqlite
# - django 4.2, python 3.12, wagtail 5.2, postgres
# - django 4.2, python 3.12, wagtail main, sqlite (allow failures)
```

99.9% of the time, the errors caught by this sort of CI setup will be either of the variety "this fails on Python 3.9 or below, because we accidentally used a Python 3.10-only language feature", or "this fails on Wagtail 5.1 and above, because Wagtail made a breaking change". The occasions when a bug only surfaces on one specific combination of Django / Python / Wagtail, or something is broken in the middle of the version range but works fine on older and newer ones, are vanishingly rare. So, there's no need to test every combination - as long as each version of Django / Python / Wagtail / database appears at least once somewhere in the list, we're well covered.

(There might be exceptions to this for more complex projects - for example, if we were doing complex ORM queries with special cases for Postgres and sqlite, it would probably be worth testing against both databases for each Django version. wagtail-webstories is definitely on the simple end of the scale though!)

Reply:
Hi @gasman ,
Thanks for the suggestion. Please see the latest commit: https://github.com/torchbox/wagtail-webstories/pull/26/commits/d0f1ae977e942d49ee9765a337fca3c37fb6b422